maplibre-navigation-core/src/androidMain/kotlin/org/maplibre/navigation/core/location/engine/GoogleLocationEngine.kt

```diff
@@ -0,0 +1,80 @@
+package org.maplibre.navigation.core.location.engine
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.os.Looper
+import android.location.Location as AndroidLocation
+import com.google.android.gms.location.FusedLocationProviderClient
+import com.google.android.gms.location.LocationListener
+import com.google.android.gms.location.LocationRequest
+import com.google.android.gms.location.LocationServices
+import com.google.android.gms.location.Priority
+import kotlinx.coroutines.channels.awaitClose
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.callbackFlow
+import org.maplibre.navigation.core.location.Location
+import org.maplibre.navigation.core.location.toLocation
+import kotlin.coroutines.resume
+import kotlin.coroutines.resumeWithException
+import kotlin.coroutines.suspendCoroutine
+
+/**
+ * A [LocationEngine] that uses the Google Play Services Location API.
+ *
+ * @param context used to initialize the [FusedLocationProviderClient]
+ * @param looper looper that is ued by the [FusedLocationProviderClient] to listen on for location updates
+ */
+open class GoogleLocationEngine(
+    context: Context,
+    private val looper: Looper
```

Comment:
This looper was nullable beforehand, which was consistent with the underlying Google API: [FusedLocationProviderClient](https://developers.google.com/android/reference/com/google/android/gms/location/FusedLocationProviderClient#requestLocationUpdates(com.google.android.gms.location.LocationRequest,%20java.util.concurrent.Executor,%20com.google.android.gms.location.LocationListener))

Since the `null` value doesn't cause any trouble and its effect is specified in Google's docs, I'd consider making it nullable again.

```suggestion
    private val looper: Looper?
```

Reply:
✅